app/src/main/java/ngo/teog/swift/gui/hospital/HospitalActivity.java

```diff
@@ -198,14 +203,45 @@ public void afterTextChanged(Editable editable) {}
             if(deviceInfos != null) {
                 int[] stateCounters = {0, 0, 0, 0, 0, 0};
 
+                Date now = new Date();
+
+                int overdueDeviceCount = 0;
+
                 for(DeviceInfo deviceInfo : deviceInfos) {
                     ReportInfo latestReportInfo = deviceInfo.getReports().get(deviceInfo.getReports().size()-1);
 
                     int state = latestReportInfo.getReport().getCurrentState();
 
                     stateCounters[state]++;
+
+                    HospitalDevice device = deviceInfo.getDevice();
+                    if (state != DeviceState.BROKEN && state != DeviceState.MAINTENANCE && state != DeviceState.IN_PROGRESS && state != DeviceState.SALVAGE) {
+                        //look for last maintenance/repair or creation
+
+                        List<ReportInfo> reversedReportInfos = deviceInfo.getReports();
+                        Collections.reverse(reversedReportInfos);
+                        Date lastMaintenance = null;
+                        for(ReportInfo reportInfo : reversedReportInfos) {
+                            Report report = reportInfo.getReport();
+                            int previousState = report.getPreviousState();
+                            int currentState = report.getCurrentState();
+
+                            if (previousState == DeviceState.MAINTENANCE || previousState == DeviceState.BROKEN || (previousState == DeviceState.WORKING && currentState == DeviceState.WORKING)) {
+                                lastMaintenance = report.getCreated();
+                                break;
+                            }
+                        }
+                        if (lastMaintenance != null) {
+                            int daysLeft = (device.getMaintenanceInterval()*7)-((int) ((now.getTime() - lastMaintenance.getTime()) / 1000 / 60 / 60 / 24));
+                            if(daysLeft <= 0){
+                                overdueDeviceCount++;
+                            }
+                        }
+                    }
                 }
 
+                overdueDevices.setText(Integer.toString(overdueDeviceCount) + " " + getString(R.string.hospital_info_overdue_devices));
```

Comment:
Conversion Integer.toString(...) is redundant when concatenating an integer with a string using the "+" - operator

Reply:
done